# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/ia2/ia2AccessibleTable.cpp
# Commit: 48dc94da9565
# Full Hash: 48dc94da9565d7f804092209a27821d729ccde60
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-04-02 09:44:13
# Regressor Bug: 1735970
# File Overlap Count: 1
# Description:
#   Bug 1735970 part 3: Add TableAccessibleBase and TableCellAccessibleBase. r=morgan
#   
#   These are needed to support RemoteAccessible tables.
#   Stuff specific to LocalAccessible is still in TableAccessible and TableCellAccessible, which now inherit from the new Base classes.
#   
# ==============================================================================

diff -r 477c59860bba -r 48dc94da9565 accessible/windows/ia2/ia2AccessibleTable.cpp
--- a/accessible/windows/ia2/ia2AccessibleTable.cpp	Fri Apr 01 09:49:56 2022 +0000
+++ b/accessible/windows/ia2/ia2AccessibleTable.cpp	Fri Apr 01 09:49:57 2022 +0000
@@ -512,7 +512,7 @@
   TableAccessible* table = TableAcc();
   if (!table) return CO_E_OBJNOTCONNECTED;
 
-  AutoTArray<LocalAccessible*, 30> cells;
+  AutoTArray<Accessible*, 30> cells;
   table->SelectedCells(&cells);
   if (cells.IsEmpty()) return S_FALSE;
 
@@ -521,8 +521,7 @@
   if (!*aCells) return E_OUTOFMEMORY;
 
   for (uint32_t i = 0; i < cells.Length(); i++) {
-    RefPtr<IAccessible> cell;
-    cells[i]->GetNativeInterface(getter_AddRefs(cell));
+    RefPtr<IAccessible> cell = MsaaAccessible::GetFrom(cells[i]);
     cell.forget(&(*aCells)[i]);
   }
 